### DISCUSSION TITLE
How to use Pickle Type

### DISCUSSION THREAD

**Discussion by RamDurgaSai** [fastapi/sqlmodel#1484] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import SQLModel, Field, PickleType



class FlowInDb(SQLModel, table=True):
    id : str = Field(primary_key=True)
    flow: PickleType
    
    class Config:
        arbitrary_types_allowed = True
```

### Description

Getting Error ... 
sqlalchemy.exc.CompileError: (in table 'flowindb', column 'flow'): Can't generate DDL for NullType(); did you forget to specify a type on this Column?

### Operating System

Windows

### SQLModel Version

0.06

### Python Version

3.10

**Comment by byrman** [comment]
Try `flow: PickleType = Field(sa_column=Column(PickleType))`.

**Comment by RamDurgaSai** [comment]
> Try `flow: PickleType = Field(sa_column=Column(PickleType))`.

Thanks 

Gettting  ... 

sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) NOT NULL constraint failed: flowindb.flow
[SQL: INSERT INTO flowindb (flow, id) VALUES (?, ?)]
[parameters: (None, 'd4pgyYaU3nVCJgXjYu6b')]
(Background on this error at: https://sqlalche.me/e/14/gkpj)

**Comment by byrman** [comment, marked as answer]
My previous answer was perhaps a bit confusing. Suppose `flow` is of type `Foo`, then your code becomes:
```
class Foo:
    pass


class FlowInDb(SQLModel, table=True):
    id : str = Field(primary_key=True)
    flow: Foo = Field(sa_column=Column(PickleType))
    
    class Config:
        arbitrary_types_allowed = True
```
Then you can insert a `FlowInDb` instance into the database like this:
```
session.add(FlowInDb(id="d4pgyYaU3nVCJgXjYu6b", flow=Foo())
session.commit()
```
You don't have to pickle yourself, PickleType takes care of that.

**Reply by YuriiMotov** [reply]
Runnable code example:

```py
from pydantic import ConfigDict
from sqlmodel import Column, Field, PickleType, Session, SQLModel, create_engine

class Foo:
    def __init__(self, a: str):
        self.a = a

class FlowInDb(SQLModel, table=True):
    model_config = ConfigDict(arbitrary_types_allowed=True)
    id : str = Field(primary_key=True)
    flow: Foo = Field(sa_column=Column(PickleType))

def main():
    engine = create_engine("sqlite:///", echo=True)
    SQLModel.metadata.create_all(engine)

    with Session(engine) as session:
        flow = FlowInDb(flow=Foo(a="asd"))
        session.add(flow)
        session.commit()

    with Session(engine) as session:
        flow_db = session.get(FlowInDb, 1)
        assert isinstance(flow_db.flow, Foo)
        assert flow_db.flow.a == "asd"
```
